packages/powersync-sdk-web/src/db/adapters/wa-sqlite/WASQLiteDBAdapter.ts

```diff
@@ -92,31 +123,34 @@ export class WASQLiteDBAdapter extends BaseObserver<DBAdapterListener> implement
   async readLock<T>(fn: (tx: LockContext) => Promise<T>, options?: DBLockOptions | undefined): Promise<T> {
     await this.initialized;
     return new Promise((resolve, reject) => {
-      // This implementation currently only uses a single connection. Locking is ensured by navigator locks
-      // TODO add concurrent connections
-      navigator.locks.request('DBLock', async () => {
-        try {
-          const res = await fn(this.generateDBHelpers({ execute: this._execute }));
-          resolve(res);
-        } catch (ex) {
-          reject(ex);
-        }
-      });
+      this.workerMethods!.acquireDBLock(
+        Comlink.proxy(async () => {
```

Comment:
Relying on automatic garbage collection via WeakRefs for Comlink proxy may work, but is generally not recommended. It would be better to explicitly call `Comlink.releaseProxy()` once the lock is released. Same for any other uses of `Comlink.proxy`.

Reply:
I initially thought that also. But from looking at the definitions and code I think that only applies for things proxied with `expose` or `wrap` which return a `Remote`. `Comlink.releaseProxy()` is not a function. `Comlink.proxy(...)` only marks the function as a proxied function.

I might be misunderstanding though. 

I have removed the need for locks to be proxied now though, the same function could be achieved solely using navigator locks.